internal/transport/transport_test.go

```diff
@@ -1431,6 +1464,38 @@ func (s) TestEncodingRequiredStatus(t *testing.T) {
 	server.stop()
 }
 
+func (s) TestHeaderHostReplacedWithResolverAddress(t *testing.T) {
```

Comment:
I would like to suggest an e2e test for this scenario instead of this test. Look at `test/authority_test.go` for an example of an e2e test which verifies authority value received at the server.

For this specific scenario, what you would need would be as follows:
- start a `StubServer` on a local port, and set the `EmptyCallF` method to run the `authorityChecker` helper function
- Use a [manual resolver](https://pkg.go.dev/google.golang.org/grpc@v1.50.1/resolver/manual#NewBuilderWithScheme) and set it up to return the address of the stub server with the `ServerName` field set.
- Call `grpc.Dial()` using the manual resolver scheme

`TestCredsHandshakeServerNameAuthority` in test/creds_test.go` contains an example of how to use the manual resolver and set the `ServerName` field.